### PR TITLE
ci: Update to match current rpm-ostree

### PR DESCRIPTION
**Pull request by cgwalters**
This copies the `ci/` directory from rpm-ostree, with much the same rationale;
among other things we don't want to depend on the Docker hub.

The specific reason I'm doing this is that I want to add a CentOS7 build, but
that means we can't use `projectatomic/ostree-tester`, and at that point we
might as well unwind it all.

### CODE REVIEW

File: ci/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+dn=$(dirname $0)
+. ${dn}/libbuild.sh
+
+install_builddeps ostree
+
+dnf install -y sudo which attr fuse gjs parallel coccinelle clang \
+    libubsan libasan libtsan PyYAML gnome-desktop-testing redhat-rpm-config \
+    elfutils ${CI_PACKAGES:-}
+
+build --enable-gtk-doc --enable-installed-tests ${CONFIGOPTS}
```
Comment (jlebon) [inline ci/build.sh]: This should probably be `${CONFIGOPTS:-}`.

File: .papr.yml [hunk 2]
```diff
@@ -4,128 +4,62 @@ branches:
...
-build:
-    config-opts: >
-      --prefix=/usr
-      --libdir=/usr/lib64
-      --enable-installed-tests=exclusive
-      --enable-gtk-doc
-      --with-curl
-      --with-openssl
+env:
+  CI_PACKAGES: "'pkgconfig(libcurl)' 'pkgconfig(openssl)'"
```
Comment (jlebon) [inline .papr.yml]: I think it should just work without the single quotes around each.

**Comment by cgwalters** [comment]
This should be good to go.

File: .papr.yml [hunk 3]
```diff
@@ -4,128 +4,64 @@ branches:
...
-# The g-d-t-r timeout is for test-pull-many.sh; if tweaking this,
-# also be sure to change the other cases below
 tests:
-    - make syntax-check
-    - ./tests/ci-commitmessage-submodules.sh
-    - make check
-    - /bin/sh -c 'gnome-desktop-testing-runner -p 0 --timeout $((10 * 60)) libostree/'
+  - ci/ci-commitmessage-submodules.sh
+  - ci/build-check.sh
+  - make syntax-check
```
Comment (jlebon) [inline .papr.yml]: Can we move `syntax-check` to run before `build-check.sh`?

File: .papr.yml [hunk 4]
```diff
@@ -4,128 +4,64 @@ branches:
...
 env:
     CFLAGS: '-fsanitize=undefined -fsanitize-undefined-trap-on-error -fsanitize=address -O2 -Wp,-D_FORTIFY_SOURCE=2'
     ASAN_OPTIONS: 'detect_leaks=0'  # Right now we're not fully clean, but this gets us use-after-free etc
     # TODO when we're doing leak checks: G_SLICE: "always-malloc"
 
-build:
-    config-opts: >
-      --prefix=/usr
-      --libdir=/usr/lib64
-      --enable-installed-tests=exclusive
```
Comment (jlebon) [inline .papr.yml]: We're losing this tidbit, so the `g-d-t-r` invocation in `build-check.sh` won't actually run anything right?

File: ci/build.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+set -xeuo pipefail
+
+dn=$(dirname $0)
+. ${dn}/libbuild.sh
+
+install_builddeps ostree
+
+dnf install -y sudo which attr fuse gjs parallel coccinelle clang \
+    libubsan libasan libtsan PyYAML gnome-desktop-testing redhat-rpm-config \
+    elfutils ${CI_PACKAGES:-}
```
Comment (jlebon) [inline ci/build.sh]: Any reason for using `CI_PACKAGES` instead of just plopping the extra ones directly in the YAML?

File: .papr.yml (hunk 4 above)
Comment (cgwalters) [inline .papr.yml]: Yes, good catch!

File: .papr.yml (hunk 3 above)
Comment (cgwalters) [inline .papr.yml]: Unfortunately `syntax-check` requires us to have run `configure` (though glancing at the code, not for good reason).

It's tempting to hook up `syntax-check` to `check`.

**Comment by cgwalters** [comment]
Fixed up for comments :arrow_up:

**Comment by jlebon** [comment]
Nice! /me makes note to delete the ostree-tester image.
@rh-atomic-bot r+ 98e3d5c

**Pull request closed without merging**